internal/cmd/beta/server/create/create.go

```diff
@@ -0,0 +1,338 @@
+	Name                          *string
+	MachineType                   *string
+	AffinityGroup                 *string
+	AvailabilityZone              *string
+	BootVolumeSourceId            *string
+	BootVolumeSourceType          *string
+	BootVolumeSize                *int64
+	BootVolumePerformanceClass    *string
+	BootVolumeDeleteOnTermination *bool
+	ImageId                       *string
+	KeypairName                   *string
+	Labels                        *map[string]string
+	NetworkId                     *string
+	NetworkInterfaceIds           *[]string
+	SecurityGroups                *[]string
+	ServiceAccountMails           *[]string
+	UserData                      *string
+	Volumes                       *[]string
+}
+
+func NewCmd(p *print.Printer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create",
+		Short: "Creates a server",
+		Long:  "Creates a server.",
+		Args:  args.NoArgs,
+		Example: examples.Build(
+			examples.NewExample(
+				`Create a server from an image with id xxx`,
+				`$ stackit beta server create --machine-type t1.1 --name server1 --image-id xxx`,
+			),
+			examples.NewExample(
+				`Create a server with labels from an image with id xxx`,
+				`$ stackit beta server create --machine-type t1.1 --name server1 --image-id xxx --labels key=value,foo=bar`,
+			),
+			examples.NewExample(
+				`Create a server with a boot volume`,
+				`$ stackit beta server create --machine-type t1.1 --name server1 --boot-volume-source-id xxx --boot-volume-source-type image --boot-volume-size 64`,
+			),
+			examples.NewExample(
+				`Create a server with a boot volume from an existing volume`,
+				`$ stackit beta server create --machine-type t1.1 --name server1 --boot-volume-source-id xxx --boot-volume-source-type volume`,
+			),
+			examples.NewExample(
+				`Create a server with a keypair`,
+				`$ stackit beta server create --machine-type t1.1 --name server1 --image-id xxx --keypair-name example`,
+			),
+			examples.NewExample(
+				`Create a server with a network`,
+				`$ stackit beta server create --machine-type t1.1 --name server1 --image-id xxx --network-id yyy`,
+			),
+			examples.NewExample(
+				`Create a server with a network interface`,
+				`$ stackit beta server create --machine-type t1.1 --name server1 --boot-volume-source-id xxx --boot-volume-source-type image --boot-volume-size 64 --network-interface-ids yyy`,
+			),
+			examples.NewExample(
+				`Create a server with an attached volume`,
+				`$ stackit beta server create --machine-type t1.1 --name server1 --boot-volume-source-id xxx --boot-volume-source-type image --boot-volume-size 64 --volumes yyy`,
+			),
+			examples.NewExample(
+				`Create a server with user data (cloud-init)`,
+				`$ stackit beta server create --machine-type t1.1 --name server1 --boot-volume-source-id xxx --boot-volume-source-type image --boot-volume-size 64 --user-data file("${path.module}/cloud-init.yaml")`,
+			),
+		),
+		PreRun: func(cmd *cobra.Command, _ []string) {
```

Comment:
I think we should make these validations consistent with similar ones in other commands, where this is always done in `parseInput` (e.g. in the [`mongodbflex instance create`](https://github.com/stackitcloud/stackit-cli/blob/3b8d284e949cc8c77281f27e107fb70e22cafbbf/internal/cmd/mongodbflex/instance/create/create.go#L169-L182) command)